### PR TITLE
Add tvOS support, nullability annotation, and Obj-C generics

### PR DESCRIPTION
**Pull request by chillpop**
@vokal/ios-developers, code review?

The diff is bad due to a pod update on the example project (using a newer version of cocoapods).

I also rearranged the podspec to group the subspecs by platform availability

### CODE REVIEW

**Comment by brockboland** [comment]
on it

File: Example/Pods/Local Podspecs/VOKUtilities.podspec.json [hunk 1]
```diff
@@ -29,40 +34,65 @@
...
+      "platforms": {
+        "ios": "7.0",
+        "tvos": "9.0"
+      },
       "source_files": "Pod/UIView+VOKCircle/*.{h,m}"
     },
     {
       "name": "UIViewController+VOKKeyboard",
+      "platforms": {
+        "ios": null
```
Comment (brockboland) [inline Example/Pods/Local Podspecs/VOKUtilities.podspec.json:79]: Do `null`s in here mean it's available for any version of iOS?

Comment (chillpop) [inline Example/Pods/Local Podspecs/VOKUtilities.podspec.json:79]: i think so but I'm not sure

Comment (brockboland) [inline Example/Pods/Local Podspecs/VOKUtilities.podspec.json:79]: I missed that this is the local podspec from CocoaPods, so it really doesn't matter

**Comment by brockboland** [comment]
Would it makes sense to (and/or is it even possible to) add some tvOS tests?

**Comment by chillpop** [comment]
It's possible to make tvOS unit tests.  As of right now, KIF doesn't support tvOS, so reusing the existing KIF UI tests isn't an option.  In testing this locally, I duplicated the existing unit tests onto a new tvOS target and they ran just fine.

I'm not sure how useful it is to re-running the same tests on tvOS though.  We aren't doing any tests on OS X despite having support for that platform too.  (That's not a good excuse to not have tests though.)  

I _could_ see the use of making sure the pods compile when linked to the tvOS SDK.

**Comment by brockboland** [comment]
That was more what I had in mind, to make sure that any code we put into the tvOS-enabled subspecs will _actually_ work on tvOS. And yes, the same is true for OS X, it's just of lesser concern for me.

**Comment by brockboland** [comment]
Talked to @chillpop offline: going to add tests in next PR

**Pull request merged**